R/mean-model.R

```diff
@@ -0,0 +1,250 @@
+      #   }
+      #   self$massBalance <- as.data.frame(self$massBalance) #Convert massBalance list to dataframe
+      # }
+
+
+      # print(paste0(self$simulationOutputsFolder,"/",self$modelDisplayName,"-outputs.csv"))
+
+      if (saveSimulation){
+        write.csv(self$simulationOutputs$data,file = paste0(self$simulationOutputsFolder,"/",self$modelDisplayName,"-outputs.csv"))
+
+        # if (calculateMassBalance){
+        #   write.csv(self$massBalance,file = paste0(self$simulationOutputsFolder,"/",self$modelDisplayName,"-mass-balance.csv"))
+        # }
+
+      }
+
+    },
+
+    calculateMassBalance = function(saveMassBalance = FALSE) {
+
+      loadedModel <- loadSimulation(self$modelFilePath, loadFromCache = FALSE)
+      #Add all outputs to loadedModel's output list
+      compoundList <- getCompoundsList(loadedModel) #Get a vector of all compound names
+      pathsPerCompoundList <-list()
+      for (cmp in compoundList){ #For loop to get the list of paths for each compound
+        pathsPerCompoundList[[cmp]]  <- getPathsForMoleculeAmount(simulation = loadedModel,cmp)
+        addOutputs(quantitiesOrPaths = pathsPerCompoundList[[cmp]], simulation = loadedModel) #add outputs for current compound to loadedModel's output list
+      }
+
+      simulationResults <- runSimulation(loadedModel)
+      simulationOutputs <- getOutputValuesTLF(simulationResults = simulationResults,
+                                              quantitiesOrPaths = simulationResults$allQuantityPaths,
+                                              individualIds = 0,
+                                              population =  loadPopulation(csvPopulationFile = "./data/popData.csv")
+      )  #using this placeholder population CSV file for now...need to make optional the population input to getOutputValuesTLF
+
+      self$massBalance <- list()
+
+      for ( cmp in compoundList ){ #For loop to calculate mass balance for each compound
+        self$massBalance[[cmp]] <- rep(0,nrow(simulationOutputs$data)) #Set up vector of zeros that is the same length as the number of simulation timepoints
+        pthsVec <- envList2PathStringsVector(pathsPerCompoundList[[cmp]]) #Get string vector of all outputs for the current compound
+        for (t in seq(1:nrow(simulationOutputs$data))){ #For each timepoint t
+          self$massBalance[[cmp]][t] <- sum(simulationOutputs$data[t,pthsVec]) # Select the columns of the dataframe self$simulationOutputs$data corresponding to the current compound and sum together the amounts of the compound in each path given in pthsVec for the timepoint t
+        }
+      }
+      self$massBalance <- as.data.frame(self$massBalance) #Convert massBalance list to datafram
+
+      # print(paste0(self$simulationOutputsFolder,"/",self$modelDisplayName,"-outputs.csv"))
+      if (saveMassBalance){
+        write.csv(self$massBalance,file = paste0(self$simulationOutputsFolder,"/",self$modelDisplayName,"-mass-balance.csv"))
+      }
+
+
```

Comment:
remove spaces

Reply:
@abdullahhamadeh Please run the `styler::tidyverse_style()` command (see install in R coding standards) on the whole package